pkg/controller/jobframework/reconciler.go

```diff
@@ -1020,3 +1022,14 @@ func clearMinCountsIfFeatureDisabled(in []kueue.PodSet) []kueue.PodSet {
 	}
 	return in
 }
+
+// FilterAnnotationsWithPrefix filters out annotations containing prefix.
+func FilterAnnotationsWithPrefix(annotations map[string]string, prefix string) map[string]string {
```

Comment:
nit: this function doesn't need to be specialized (drop the second parameter). rename accordingly